nostr-client-plus/src/bin/aggregator.rs

```diff
@@ -403,6 +411,238 @@ async fn run() -> Result<()> {
     Ok(())
 }
 
+async fn finalize_classification(
+    assign_event_id: String,
+    payload: &ResultPayload,
+    event_id: String,
+    collection: &Collection<AssignerTask>,
+    finish_job_collection: &Collection<FinishedJobs>,
+) -> Result<()> {
+    tracing::debug!("Finalizing classifications for event: {}", event_id);
+    let filter = doc! {
+        "event_id": event_id.clone(),
+    };
+
+    let mut cursor = collection
+        .find(filter, None)
+        .await?;
+    let mut tasks: Vec<AssignerTask> = Vec::new();
+    while let Some(task) = cursor.try_next().await? {
+        tasks.push(task);
+    }
+
+    let mut processing_count = 0;
+    let mut answers: HashMap<String, i32> = HashMap::new();
+    for task in tasks.iter() {
+        tracing::debug!("Worker id: {}", hex::encode(task.worker.to_bytes()));
+        if task.result.is_empty() && task.status == AssignerTaskStatus::Pending {
+            processing_count += 1;
+            break;
+        }
+        let result_identifier = match task.get_result_identifier() {
+            Ok(val) => {
+                val
+            }
+            Err(err) => {
+                return Err(anyhow!("Failed to find result identifier for task: {}", task.event_id));
```

Comment:
Also, why do we return? If there is one task with a bad identifier, are we throwing everything away?